rollup/tracing/tracing.go

```diff
@@ -579,24 +582,27 @@ func (env *TraceEnv) fillBlockTrace(block *types.Block) (*types.BlockTrace, erro
 			CodeSize:         statedb.GetCodeSize(env.coinbase),
 		},
 		Header:            block.Header(),
+		Bytecodes:         make([]*types.BytecodeTrace, 0, len(env.Codes)),
 		StorageTrace:      env.StorageTrace,
 		ExecutionResults:  env.ExecutionResults,
 		TxStorageTraces:   env.TxStorageTraces,
 		Transactions:      txs,
 		StartL1QueueIndex: env.StartL1QueueIndex,
 	}
 
-	for i, tx := range block.Transactions() {
-		evmTrace := env.ExecutionResults[i]
-		// Contract is created.
-		if tx.To() == nil {
-			evmTrace.ByteCode = hexutil.Encode(tx.Data())
-		} else { // contract call be included at this case, specially fallback call's data is empty.
-			evmTrace.ByteCode = hexutil.Encode(statedb.GetCode(*tx.To()))
-			// Get tx.to address's code hash.
-			codeHash := statedb.GetPoseidonCodeHash(*tx.To())
-			evmTrace.PoseidonCodeHash = &codeHash
-		}
+	blockTrace.Bytecodes = append(blockTrace.Bytecodes, &types.BytecodeTrace{
+		CodeSize:         0,
+		KeccakCodeHash:   codehash.EmptyKeccakCodeHash,
+		PoseidonCodeHash: codehash.EmptyPoseidonCodeHash,
+		Code:             hexutil.Encode([]byte{}),
+	})
+	for _, codeInfo := range env.Codes {
+		blockTrace.Bytecodes = append(blockTrace.Bytecodes, &types.BytecodeTrace{
+			CodeSize:         codeInfo.CodeSize,
+			KeccakCodeHash:   codeInfo.KeccakCodeHash,
+			PoseidonCodeHash: codeInfo.PoseidonCodeHash,
+			Code:             hexutil.Encode(codeInfo.Code),
```

Comment:
I think we can change `BytecodeTrace` type definition to
```
type BytecodeTrace struct {
	CodeSize         uint64        `json:"codeSize"`
	KeccakCodeHash   common.Hash   `json:"keccakCodeHash"`
	PoseidonCodeHash common.Hash   `json:"hash"`
	Code             hexutil.Bytes `json:"code"`
}
```
so that we don't need to do `hexutil.Encode` here

Reply:
fixed